File: frontend/src/types/activity.ts
export type RunStatus = "success" | "partial" | "failed" | "running";

export interface ScrapeRun {
  id: string;
  sourceId: string;
  sourceName: string;
  status: RunStatus;
  itemsFound: number;
  itemsNew: number;
  itemsDeduped: number;
  durationMs: number;
  errorMessage: string | null;
  startedAt: string;
  finishedAt: string | null;
}

export interface ActivityStats {
  totalRuns: number;
  successRate: number;
  itemsCollectedToday: number;
  activeSourceCount: number;
}

File: backend/schemas/activity.py
"""Activity and scrape run schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ScrapeRunOut(BaseModel):
    id: str
    source_id: str | None
    source_name: str | None = None
    scrape_job_id: str | None = None
    status: str
    items_found: int
    items_new: int
    items_deduped: int
    duration_ms: int
    error_message: str | None
    started_at: datetime
    finished_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class ActivityStats(BaseModel):
    total_runs: int
    success_rate: int
    items_collected_today: int
    active_source_count: int
